### PR TITLE
add device array reshape

### PR DESCRIPTION
**Pull request by omlins**

### CODE REVIEW

File: src/device/gcn/array.jl [hunk 1]
```diff
@@ -133,3 +133,20 @@ end
         nothing
     end
 end
+
+## reshape
+
+function Base.reshape(a::ROCDeviceArray{T,M,A}, dims::NTuple{N,Int}) where {T,N,M,A}
+#   if prod(dims) != length(a)
```
Comment (pxl-th) [inline src/device/gcn/array.jl:140]: Proper exception throwing requires fixes to work on Julia 1.11+.
For now I think we can do `throw(nothing)` to at least signal an error that silently skipping it.

**Review by pxl-th — APPROVED** [review]
Minor comment, other than that looks good!

**Comment by omlins** [comment]
@pxl-th can this be merged and released? Thanks and merry christmas! ;)

**Comment by pxl-th** [comment]
Yes, definitely.
Thank you! And merry christmas to you! 🎄🎄🎄

**Pull request merged**